samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/AssessmentGradingFacadeQueries.java

```diff
@@ -3398,7 +3398,7 @@ public void autoSubmitAssessments() {
 						" where a.publishedAssessmentId = c.assessment.publishedAssessmentId " +
 						" and c.retractDate <= ?" +
 						" and a.status not in (5) and (a.hasAutoSubmissionRun = 0 or a.hasAutoSubmissionRun is null) and c.autoSubmit = 1 " +
-						" and a.submittedDate is not null and a.attemptDate <= c.retractDate " +
+						" and a.submittedDate is not null and (a.attemptDate <= c.retractDate or c.lateHandling = 2) " +
```

Comment:
Hi Daniel,

Yes, I think your revised SQL works. NOT_ACCEPT_LATE_SUBMISSION == 2 and that means the user cannot submit after the due date.